src/OrchardCore/OrchardCore.DisplayManagement/Views/ShapeResult.cs

```diff
@@ -104,6 +105,12 @@ private async Task ApplyImplementationAsync(BuildShapeContext context, string di
                 return;
             }
 
+            // If a condition has been applied to this result evaluate it only if the shape has been placed.
+            if (_ifAsync != null && !(await _ifAsync.Invoke()))
```

Comment:
Could we use `!(await _ifAsync?.Invoke())` instead?

Reply:
That's okay because we would need something like this

    await ( _ifAsync?.Invoke() ?? Task.FromResult(false) )

Here I suggest

    if (_ifAsync != null && !await _ifAsync())